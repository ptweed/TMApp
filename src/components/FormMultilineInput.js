import React, {useEffect, useState} from 'react'
import { Platform, Dimensions, StyleSheet, View, TextInput, Keyboard } from 'react-native'
import { FontAwesome5 } from '@expo/vector-icons'
import { isTablet } from './Platform'

const FormMultilineInput = ({
  iconName,
  iconColor,
  name,
  placeholder,
  ...rest
}) => {

  const [calcWidth, setCalcWidth] = useState(250)
  useEffect(() => {
    if (isTablet()) setCalcWidth(Dimensions.get('screen').width - 130)

    const onChange = () => {
      if (isTablet()) setCalcWidth(Dimensions.get('screen').width - 130)
    };
    
    Dimensions.addEventListener('change', onChange)
    return () => Dimensions.removeEventListener('change', onChange);
  }, [])

  return (
  <View style={styles.inputContainer}>
    <FontAwesome5 name={iconName} size={28} color={iconColor} style={styles.iconStyle} />
    <TextInput
      {...rest}
      placeholderTextColor='grey'
      name={name}
      placeholder={placeholder} 
      multiline={true}
      numberOfLines={4}
      style={{fontSize: 17, maxHeight: 150, width: calcWidth }}
      blurOnSubmit={true}
      onSubmitEditing={()=>{Keyboard.dismiss()}}
      // keyboardType="default"
      returnKeyType="next"
      autoCapitalize="sentences"
    />
  </View>
)
}

const styles = StyleSheet.create({
  inputContainer: {
    marginRight: 25,
    marginLeft: 25, 
    marginTop: 15,
    marginBottom: 0,
    flexDirection: 'row',
    borderBottomWidth: 1
  },
  iconStyle: {
    marginRight: 5, 
    ...Platform.select({
      ios: {
        marginBottom: 5, 
        marginTop: 0,
      },
      android: {
        marginBottom: 5, 
        marginTop: 25,
      },
      default: {
        // other platforms, web for example
        marginBottom: 5, 
        marginTop: 25,
      }
    })
  }
})

export default FormMultilineInput