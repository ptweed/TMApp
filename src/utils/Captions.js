export default Captions = {
    login: "LOGIN",
    emailLabel: "Email",
    emailPlaceholder: "Enter email",
    emailValidation: "Enter a valid email",
    emailRequired: "Please enter a registered email",
    passwordLabel: "Password",
    passwordPlaceholder: "Enter password",
    passwordRequired: "Password is a required field",
    passwordValidation: "Must Contain 8 Characters, One Uppercase, One Lowercase, One Number and One Special Case Character",
    signUpForAccount: "Don't have an account? Sign Up",
    signUpForAccountConfirm: "Enter sign up verification code",
    forgottenPassword: "Forgotten password",
    accountNotVerifiedYet: "Account not verified yet",
    existingUser: "Existing user found. Please reset your password",
    incorrectEmailOrPassword: "Incorrect email or password",
    userDoesNotExist: "User does not exist. Please sign up",
    sendCode: "SEND CODE",
    backToLogin: "Back to Login",
    verificationCodeLabel: "Verification code",
    verificationCodeRequired: "Please enter your verification code",
    verificationCodePlaceholder: "Enter verification code",
    verificationCodeDoesNotExist: "Incorrect or expired verification code\n\nCode expires after 24 hours",
    verificationCodeAlreadyConfirmed: "User has already been confirmed\n\nPlease log in",
    confirmPasswordLabel: "Confirm password",
    confirmPasswordRequired: "Please confirm your password",
    confirmPasswordPlaceholder: "Confirm password",
    confirmPasswordVerification: "Passwords must match",
    confirmationCodePlaceholder: "Enter verification code",
    confirm: "CONFIRM",
    verify: "VERIFY",
    resendVerificationCode: "Resend Verification Code",
    passwordsDoNotMatch: "Passwords do not match!",
    signUp: "SIGN UP",
    tagline: "Organization on the go",
    infrastructureError: "Our apologies, our infrastructure provider is experiencing issues.\n\nPlease try using the application later.",
    errorApology: "Apologies, we have experienced an error",
    appName: "Task Master",
    loginUnavailable: "No internet connection - login unavailable",
    editingUnavailable: "No internet connection - no editing available",
    logout: "Logout",
    newTask: "New Task",
    editTask: "Edit Task",
    taskLabel: "Task",
    taskPlaceholder: "Enter task",
    taskRequired: "Please enter your task",
    taskSave: "SAVE",
    taskCreate: "CREATE",
    taskCancel: "Cancel",
    dueDateLabel: "Due date",
    dueDatePlaceholder: "Select due date",
    dueDateRequired: "Please enter the task due date",
    dueDateConfirm: "Confirm",
    dueDateCancel: "Cancel",
    noTasksGetStarted: "You have no tasks!\n\nCreate a task to get started...",
    noLateTasks: "You have no late tasks...\n\nwell done!",
    todoTasksTitle: "To do tasks:",
    lateTasksTitle: "Late tasks:",
    allTasksTitle: "All tasks - completed & late:"
  };