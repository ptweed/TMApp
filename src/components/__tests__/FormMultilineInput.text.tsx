import React from 'react'
import { render } from 'react-native-testing-library'
import Captions from '../../utils/Captions'
import FormMultilineInput from '../FormMultilineInput'

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: ''
}))
  
describe('FormMultilineInput Component', () => {
  it('FormMultilineInput renders correctly', () => {

    const formI = render(<FormMultilineInput
        name='task'
        placeholder={Captions.taskPlaceholder}
        autoCapitalize='none'
        iconName='edit'
        iconColor='#2C384A'
/>
    ).toJSON();
    expect(formI).toMatchSnapshot();

  });

});