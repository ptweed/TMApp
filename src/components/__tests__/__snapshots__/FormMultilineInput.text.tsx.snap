// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormMultilineInput Component FormMultilineInput renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "marginBottom": 0,
      "marginLeft": 25,
      "marginRight": 25,
      "marginTop": 15,
    }
  }
>
  <
    color="#2C384A"
    name="edit"
    size={28}
    style={
      Object {
        "marginBottom": 5,
        "marginRight": 5,
        "marginTop": 0,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    blurOnSubmit={true}
    keyboardType="default"
    multiline={true}
    name="task"
    numberOfLines={4}
    onSubmitEditing={[Function]}
    placeholder="Enter task"
    placeholderTextColor="grey"
    rejectResponderTermination={true}
    returnKeyType="next"
    style={
      Object {
        "fontSize": 17,
        "maxHeight": 150,
        "width": 620,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
